web-browsers/the-dom-and-html.md

```diff
@@ -0,0 +1,122 @@
+| DOM     | A representation of the content of a webpage in the form of a tree data structure. The root node is an object that represents the document, or the entire web page. | Document Object Model     | "The DOM shows us how content on a web page has nested relationships with each other."                                                                            |
+| Tree    | A data structure that resembles an upside down tree. Each piece of data is represented by a node, beginning with a root node.                                       | -                         | "The DOM is in a tree structure because it describes content nodes that point to other content nodes," "Tree data structures are used in some search algorithms." |
+| Node    | A unit of data in a tree, which contains the data itself and a list of other nodes it points to. Each node can only have one parent node.                           | Branch or Leaf            | "A node can point to other nodes, which make them children nodes," "All nodes have a parent node except for the root node."                                       |
+| HTML    | A markup language used to structure the content of a web page.                                                                                                      | Hypertext Markup Language |
+| Element | A unit of content in an HTML document. These are represented as nodes in the DOM.                                                                                   | HTML Element              |
+| Sibling | In a tree structure, a sibling node is a node that shares a parent with another node                                                                                |
+
+### !callout-info
+
+## Web Pages Are Documents
+
+Some contexts may refer to a web page as a document.
+
+### !end-callout
+
+## The DOM
+
+The Document Object Model, also known as **the DOM**, is a representation of the structured content on a web page.
+
+The DOM is a _tree structure_. Each part of the document (or web page) is represented as a node in the tree.
+
+### Trees
+
+Trees are data structures which organize data through nodes and branches.
+
+There is one root node. The root node has children nodes.
+
+Those children nodes can also have their own child nodes! However, each node can either have exactly zero or one parent node.
+
+![A tree structure with grey, unlabeled nodes. The root node has two children. Each node has either 0, 1, or 2 children. The tree is four layers deep.](../assets/web-browsers_the-dom-and-html_generic-tree.png)
+
+When a web page is represented in the DOM structure, the root node of the tree is always a document object that represents the document itself.
+
+Elements and pieces of content become children nodes to the document.
```

Comment:
the sense of change of "become" feels odd to me. I'd tend to say they are represented as descendants (descendants rather than _child_ nodes since not all elements in the tree are child nodes of the root node itself).
This suggestion assumes the above change was taken.
```suggestion
Other page elements and pieces of content are represented as descendants of the root node.
```
Here's a version that does _not_ assume the above change was taken.
```suggestion
Page elements and pieces of content are represented as descendants of the root node.
```